packages/presentation/lib/screen/home/bloc/home_bloc.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:math';
+
+import 'package:domain/usecase/generate_number_usecase.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:domain/usecase/check_number_usecase.dart';
+
+part 'home_event.dart';
+
+part 'home_state.dart';
+
+class HomeBloc extends Bloc<HomeEvent, HomeState> {
+  final blocCheckUseCase = CheckNumberUseCase();
+  final blocGenerateUseCase = GenerateNumberUseCase();
+  final Random random = Random();
+
+  HomeBloc() : super(const HomeState()) {
+    on<GenerateNumberEvent>(_generate);
+    on<SubmitNumberEvent>(_submit);
+    on<ChangeSubmitButtonStateEvent>(_changeSubmitButton);
+  }
+
+  void _generate(
+    GenerateNumberEvent event,
+    Emitter<HomeState> emit,
+  ) {
+    final generatedNumber = blocGenerateUseCase.call();
+    emit(
+      state.copyWith(
+        generatedNumber: generatedNumber,
+        attempts: 0,
+        gameState: GameState.inGame,
+      ),
+    );
+  }
+
+  void _submit(
+    SubmitNumberEvent event,
+    Emitter<HomeState> emit,
+  ) {
+    if (state.generatedNumber != null) {
+      final checkParams = CheckNumberParams(
+        state.generatedNumber!,
+        event.predictedNumber,
+      );
+      final isNumberGuessed = blocCheckUseCase.call(checkParams);
+      final attempts = state.attempts + 1;
```

Comment:
mb better to move incrementation in to domain

Reply:
No need

Reply:
it's ok